modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/UserNotificationTypeTest.java

```diff
@@ -86,6 +91,80 @@ public void tearDown() throws PortalException {
 			user2.getUserId());
 	}
 
+	@Test
+	public void testIsDelivery() throws Exception {
+		NotificationTemplate notificationTemplate =
+			notificationTemplateLocalService.addNotificationTemplate(
+				_createNotificationContext(
+					Arrays.asList(
+						NotificationRecipientSettingUtil.
+							createNotificationRecipientSetting(
+								"userScreenName", user1.getScreenName())),
+					NotificationRecipientConstants.TYPE_USER));
+
+		ObjectAction objectAction = objectActionLocalService.addObjectAction(
+			RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+			childObjectDefinition.getObjectDefinitionId(), true,
+			StringPool.BLANK, RandomTestUtil.randomString(),
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			RandomTestUtil.randomString(),
+			ObjectActionExecutorConstants.KEY_NOTIFICATION,
+			ObjectActionTriggerConstants.KEY_ON_AFTER_UPDATE,
+			UnicodePropertiesBuilder.put(
+				"notificationTemplateId",
+				notificationTemplate.getNotificationTemplateId()
+			).build(),
+			false);
+
+		_executeNotificationObjectAction(
+			notificationTemplate, childObjectDefinition, group.getGroupKey(),
+			user1);
+
+		List<NotificationQueueEntry> notificationQueueEntries =
+			notificationQueueEntryLocalService.getNotificationQueueEntries(
+				QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+		Assert.assertEquals(
+			notificationQueueEntries.toString(), 1,
+			notificationQueueEntries.size());
+
+		// Change UserNotificationDeliver
+
+		UserNotificationDelivery userNotificationDelivery =
+			_userNotificationDeliveryLocalService.fetchUserNotificationDelivery(
+				TestPropsValues.getUserId(),
+				childObjectDefinition.getPortletId(),
+				_classNameLocalService.getClassNameId(
+					childObjectDefinition.getClassName()),
+				UserNotificationDefinition.NOTIFICATION_TYPE_UPDATE_ENTRY,
+				UserNotificationDeliveryConstants.TYPE_WEBSITE);
+
+		_userNotificationDeliveryLocalService.updateUserNotificationDelivery(
+			userNotificationDelivery.getUserNotificationDeliveryId(), false);
+
+		_executeNotificationObjectAction(
+			notificationTemplate, childObjectDefinition, group.getGroupKey(),
+			user1);
+
+		notificationQueueEntries =
+			notificationQueueEntryLocalService.getNotificationQueueEntries(
+				QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+		Assert.assertEquals(
+			notificationQueueEntries.toString(), 1,
+			notificationQueueEntries.size());
+
+		objectActionLocalService.deleteObjectAction(
+			objectAction.getObjectActionId());
+		_userNotificationEventLocalService.deleteUserNotificationEvents(
+			user1.getUserId());
+		notificationQueueEntryLocalService.deleteNotificationQueueEntry(
+			notificationQueueEntries.get(
+				0
+			).getNotificationQueueEntryId());
```

Comment:
@mariuo here you just need to delete the notification queue entry:

````java
notificationQueueEntryLocalService.deleteNotificationQueueEntry(notificationQueueEntries.get(0))
````

Reply:
the object action and the user notification event will be deleted by another engine called [DataGuard](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/portal-test/src/com/liferay/portal/kernel/test/rule/DataGuardTestRule.java#L13) after the test running.